app/models/environment.rb

```diff
@@ -45,6 +56,16 @@ def locked?
     active_lock.present?
   end
 
+  # Returns the first lock in the queue, or nil
+  def next_in_line
+    locks.waiting.first
```

Comment:
I think we should probably add an explicit `order` here. I'm not sure the order is deterministic if it's not specified.